test/gtest/uct/test_rc_flow_control.cc

```diff
@@ -0,0 +1,76 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
+* Copyright (C) UT-Battelle, LLC. 2016. ALL RIGHTS RESERVED.
+* Copyright (C) ARM Ltd. 2016.All rights reserved.
+* See file LICENSE for terms.
+*/
+
+extern "C" {
+#include <uct/api/uct.h>
+#include <uct/ib/rc/base/rc_ep.h>
+#include <uct/ib/rc/base/rc_iface.h>
+}
+#include <common/test.h>
+#include "uct_test.h"
+
+class test_rc_flow_control : public uct_test {
+public:
+    void initialize() {
+        uct_test::init();
+
+        m_e1 = uct_test::create_entity(0);
+        m_entities.push_back(m_e1);
+
+        m_e2 = uct_test::create_entity(0);
+        m_entities.push_back(m_e2);
+    }
+
+    void connect() {
+        m_e1->connect(0, *m_e2, 0);
+        m_e2->connect(0, *m_e1, 0);
+    }
+
+    uct_rc_iface_t* rc_iface(entity *e) {
+        return ucs_derived_of(e->iface(), uct_rc_iface_t);
+    }
+
+    static ucs_status_t ib_am_dummy_handler(void *arg, void *data, size_t length, void *desc) {
+        return UCS_OK;
+    }
+
+    void cleanup() {
```

Comment:
please add 'virtual' here because you're overriding a virtual function, your function would be virtual anyway so it would be better to "document" this